MueLu: Adding parameters to control performance and tracing of AMG

seanofthemillers opened the pull request:
<!---
Be sure to select `develop` as the `base` branch against which to create this
pull request.  Only pull requests against `develop` will undergo Trilinos'
automated testing.  Pull requests against `master` will be ignored.

Provide a general summary of your changes in the Title above.  If this pull
request pertains to a particular package in Trilinos, it's worthwhile to start
the title with "PackageName:  ".

Note that anything between these delimiters is a comment that will not appear
in the pull request description once created. Most areas in this message are
commented out and can be easily added by removing the comment delimiters.

Please make sure to mark:
* Reviewers
* Assignees
* Labels

Replace <teamName> below with the appropriate Trilinos package/team name.
-->
@trilinos/muelu

## Motivation
<!--- 
Why is this change required?  What problem does it solve? Please link to a github 
issue that describes the problem/issue/bug this PR solves.
-->

<!---
If applicable, let us know how this merge request is related to any other open
issues or pull requests:

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of 
-->

These changes address three things:
- MueLu: Adding parameter for aggregation in AMG to reuse coloring results from a previous iteration.
- Ifpack2: Adding parameter to use native SpMV/blas kernels instead of fused SpMV kernel for applying Chebyshev smoother (noticeable performance difference on GPUs with vendor libraries).
- Tpetra: Adding fix for Kokkos profiling regions, which can otherwise cause incorrect results with tracing tools.


## Stakeholder Feedback
<!--- 
If a github issue includes feedback from the relevant stakeholder(s), please link it.  
If the stakeholder(s) communicated that feedback through a different medium, please note that you did so.
-->

## Testing
<!---
Please confirm that any classes or functions in the Trilinos library that this PR touches are 
exercised by at least one test in Trilinos.  Please specify which test that is.  For untestable 
changes (e.g. changes to the nightly testing system) or changes to Trilinos tests, please say "N/A".

-->

<!--- 
## Additional Information
Anything else we need to know in evaluating this merge request?
 -->

cgcgcg commented:
@seanofthemillers  Where is this actually reusing the coloring?

cgcgcg commented:
We might want to integrate the reuse of coloring in the existing reuse mechanism. I'm not sure right now if that works well with our current string option.

seanofthemillers commented:
> @seanofthemillers Where is this actually reusing the coloring?
Ha, I missed a copy/paste somewhere. I'll fix that it a moment.

cgcgcg commented:
@trilinos/muelu

jhux2 commented on packages/ifpack2/src/Ifpack2_Details_Chebyshev_def.hpp:
```diff
@@ -481,6 +485,10 @@ setParameters (Teuchos::ParameterList& plist)
 
+  // Load the kernel fuse override from the parameter list
+  if (plist.isParameter ("chebyshev: use native spmv"))
+    ckUseNativeSpMV = plist.get("chebyshev: use native spmv", ckUseNativeSpMV);
```
This should also be added to the list of allowable Ifpack2 options, in `Ifpack2_Parameters.cpp`.

seanofthemillers replied:
Will do.

seanofthemillers replied:
Fixed.

jhux2 reviewed (commented):
@seanofthemillers Thanks for the contribution.  See my comment.

seanofthemillers commented on packages/muelu/src/Utils/MueLu_Monitor.cpp:
```diff
@@ -4,6 +4,7 @@ int MueLu::FactoryMonitor::timerIdentifier_=0;
 namespace MueLu {
   PrintMonitor::PrintMonitor(const BaseClass& object, const std::string& msg, MsgType msgLevel) : object_(object) {
     tabbed = false;
+    Kokkos::Profiling::pushRegion(msg);
```
Ah, I wasn't seeing any MueLu tracing regions in my analysis, and couldn't find a flag to enable it, so I just dropped this into the Factory/SubMonitor via the PrintMonitor. I can remove this line since it feels a bit out of place. Is there a flag/env-var that enables kokkos tracing in MueLu?

cgcgcg replied:
You could try `Teuchos_KOKKOS_PROFILING=ON`.

jhux2 replied:
@seanofthemillers I've been able to get traces before.  Do have the Kokkos profiling tools enabled?

seanofthemillers replied:
I have removed this line. I have the `Teuchos_KOKKOS_PROFILING` on and tested with the roctx tracer from the kokkos-tools repo, but was not able to get MueLu factories to show up in the trace (I had lots of Belos and Tpetra regions though). I'm probably missing something that enables kokkos profiling in Teuchos's timers - I'll look at that the next chance I get.

seanofthemillers replied:
Ah, figured it out. I needed to set the verbosity flag to "timings" in the preconditioner file in order to enable the Teuchos stacked timer in the TimeMonitor, which then triggers the kokkos profiling regions.

cgcgcg commented on packages/muelu/src/Graph/UncoupledAggregation/MueLu_UncoupledAggregationFactory_kokkos_def.hpp:
```diff
@@ -284,79 +304,109 @@ namespace MueLu {
+        
+      TEUCHOS_TEST_FOR_EXCEPTION(reuse_coloring && (not pL.get<bool>("aggregation: deterministic")), 
+                                 Exceptions::RuntimeError, 
+                                 "MueLu::UncoupledAggregationFactory::Build: Request to reuse coloring requires aggregation to be deterministic!");
```
Why do we need deterministic aggregation for color reuse?

seanofthemillers replied:
Yup, it appears that it will change the coloring each iteration, so we need it to be deterministic. I assume. I get segfaults if I allow coloring reuse with non-deterministic.

cgcgcg replied:
I don't think so. We just need valid D2 coloring. The non-deterministic one should also work unless I'm missing something.

seanofthemillers replied:
Still tracking this down. Currently this error appears to be tied to the reuse of the level 1 aggregation algorithm `AggregationPhase1Algorithm_kokkos::BuildAggregatesRandom` (and likely others), which results in a `numNonAggragatedNodes<0` condition.

cgcgcg replied:
Wait, numNonAggregatedNodes is actually negative?

seanofthemillers replied:
I see. Is there an easy way to check if the graph has changed (without storing the previous graph)?

cgcgcg replied:
Well, we are calling the aggregation because we want to generate a new graph. So I think the assumption should be that it changes.

seanofthemillers commented:
@cgcgcg You mentioned that I should change the name from "aggregation: reuse coloring" = ON/OFF. Would "reuse: type" = "coloring" be preferable?

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

cgcgcg commented:
@seanofthemillers Something along those lines, but we need to double-check that this isn't orthogonal to "reuse: type".

seanofthemillers commented:
@jhux2 I would be happy to, but I'm curious, is there a way to check if these options are being used? As it stands, the solver will work fine if the option is ignored since this is more of a performance improvement. I'll get started on a test.

seanofthemillers commented:
@cgcgcg @jhux2 I have addressed your comments (I think) and have added a new test, though it is a little limited. I have used a tracing API to confirm that the flags are indeed being obeyed, but the flags become ignored at some point, the test will likely still pass. Could you give this another look?

cgcgcg commented:
@seanofthemillers We will. Just to better understand: what is the motivation for keeping aggregate information when you can keep tentative P instead?

seanofthemillers commented:
@cgcgcg Ah, this might just be me misunderstanding what tentative P is. For the problem I'm looking at, I am assuming that the values stored in matrix A are changing on each re-use, but sparsity of A is remaining the same. When I first looked through the factories, I saw that the aggregation was purely topology based, while tentative P contained values. Are the values in tentative P are just defined from the aggregation/topology or do they depend on the values in A? Ptent assembly seems to require the Nullspace of A, and has some QR factorization component (based on the trace), so I assumed it depended on A's values.

cgcgcg commented:
Does you use case involve changing the Nullspace? Tentative P does not use the values of A.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

cgcgcg commented:
You could cherry-pick d1dc49e76675b83d104573a60eb3ddf42ee21d44 from PR #11100. (I'm having some issues getting that PR merged just yet.)

csiefer2 commented:
@seanofthemillers What's the logic on the Tpetra changes?  Something that corrects for people enabling and disabling profiling within the life of a profiling region?

seanofthemillers commented:
@csiefer2 The Tpetra changes are there because the ProfilingRegion can enable "groups" of profiling regions. If a group is not enabled, then the ProfilingRegion class will still pop the top region in its destructor. These changes make it so it only pops a region if a region is active.

csiefer2 commented:
@cgcgcg From my inspection, the seems to be passing now on geminga.  Do you concur?

csiefer2 commented:
@seanofthemillers Not following.  The existing code only pushes/pops a region if Behavior::profilingRegionUseKokkosProfiling() is true.  Is there something wrong in the logic of that function?

seanofthemillers commented:
@csiefer2 Ah, take a look at the `timeit` variable as part of the `ProfilingRegion (const char name[], const char group[])` constructor. It will turn off the profiling region if the `group` is not enabled, but it will always pop the top region in the destructor independent of a group being enabled.

cgcgcg commented:
@seanofthemillers  I'd suggest to remove the MueLu changes and merge the Tpetra and Ifpack2 stuff.

seanofthemillers commented:
@cgcgcg Will do, I'll remove the MueLu changes.

seanofthemillers commented:
@cgcgcg MueLu changes have been removed.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - User Requested Retest - Label AT: RETEST will be reset after testing.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED by label AT: PRE-TEST INSPECTED! Autotester is Removing Label; This inspection will remain valid until a new commit to source branch is performed.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0</h3>
<ul>
  <li>Build Num: 1242</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-8.3.0-openmpi-1.10.1-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-7.2.0-serial</h3>
<ul>
  <li>Build Num: 1242</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-7.2.0-debug</h3>
<ul>
  <li>Build Num: 1239</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-openmpi-1.10.1-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_intel-17.0.1</h3>
<ul>
  <li>Build Num: 1239</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-intel-17.0.1-mpich-3.2-serial_release-debug_static_no-kokkos-arch_no-asan_no-complex_fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_clang-10.0.0</h3>
<ul>
  <li>Build Num: 1255</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-clang-10.0.0-openmpi-1.10.1-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-10.1.243</h3>
<ul>
  <li>Build Num: 1257</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>ats2_cuda-10.1.243-gnu-8.3.1-spmpi-rolling_release_static_Volta70_Power9_no-asan_no-complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>vortex</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_python3</h3>
<ul>
  <li>Build Num: 1047</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-anaconda3-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_pr-framework</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>ascic</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-11.4.2-uvm-off</h3>
<ul>
  <li>Build Num: 776</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>GPU</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (seanofthemillers/Trilinos)</dt>
  <ul>
    <li>Branch: muelu_additional_control_parameters</li>
    <li>SHA: 0983388aa182899b969f59c9ef5438ed09412f8e</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: seanofthemillers
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0</h3>
<ul>
  <li>Build Num: 1242</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-8.3.0-openmpi-1.10.1-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-7.2.0-serial</h3>
<ul>
  <li>Build Num: 1242</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-7.2.0-debug</h3>
<ul>
  <li>Build Num: 1239</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-openmpi-1.10.1-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_intel-17.0.1</h3>
<ul>
  <li>Build Num: 1239</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-intel-17.0.1-mpich-3.2-serial_release-debug_static_no-kokkos-arch_no-asan_no-complex_fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_clang-10.0.0</h3>
<ul>
  <li>Build Num: 1255</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-clang-10.0.0-openmpi-1.10.1-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-10.1.243</h3>
<ul>
  <li>Build Num: 1257</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>ats2_cuda-10.1.243-gnu-8.3.1-spmpi-rolling_release_static_Volta70_Power9_no-asan_no-complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>vortex</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_python3</h3>
<ul>
  <li>Build Num: 1047</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-anaconda3-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_pr-framework</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>ascic</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-11.4.2-uvm-off</h3>
<ul>
  <li>Build Num: 776</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: Tpetra;pkg: MueLu;AT: RETEST</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>11113</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>GPU</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>muelu_additional_control_parameters</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/seanofthemillers/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>0983388aa182899b969f59c9ef5438ed09412f8e</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a76c1c4a948a24f6e0728da96367e0d8d5017442</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 11113.](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-11113-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           THE LAST COMMIT TO THIS PULL REQUEST HAS NOT BEEN REVIEWED YET!

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ cgcgcg ]!

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Merge on Pull Request# 11113: IS A SUCCESS - Pull Request successfully merged
